### PR TITLE
fix: [Debug] Backtrace does not display contents of file in PHP +8.3.0

### PR DESCRIPTION
**Pull request by Nightprince**
https://www.php.net/manual/en/function.highlight-string.php

**Checklist:**
- [x] Securely signed commits
- [ ] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [ ] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

**Comment by kenjis** [comment]
Thank you for sending PR!

Can you explain what's the issue, and add test code to prove your fix?

File: system/Debug/Exceptions.php [hunk 1]
```diff
@@ -541,9 +541,12 @@ public static function highlightFile(string $file, int $lineNumber, int $lines =
             return false;
         }
 
+        $version = explode('.', PHP_VERSION);
+        $php_version = (int)$version[0] * 10000 + (int)$version[1] * 100 + (int)$version[2];
```
Comment (kenjis) [inline system/Debug/Exceptions.php:545]: This function is deprecated, and no longer used.
Do you really need to fix this function?
If so, use `PHP_VERSION_ID` instead. See `BaseExceptionHandler::highlightFile()`

**Comment by paulbalandan** [comment]
I think this is fixing the highlight_string output which I believe has been fixed already in 4.5 branch.

**Comment by kenjis** [comment]
I think it was fixed in #8401.

@Nightprince Can you show the steps to reproduce the issue?
Why do you need to fix the deprecated method?

**Comment by datamweb** [comment]
@Nightprince Can you post a screenshot before and after this PR?

**Comment by Nightprince** [comment]
> @Nightprince Can you post a screenshot before and after this PR?

Before: 

Screenshot 2024-03-14 095340

After:

Screenshot 2024-03-14 095510

**Comment by kenjis** [comment]
It seems you are using very old CI4. Could you update to v4.4.6?

**Comment by datamweb** [comment]
@Nightprince As mentioned, your CI version is outdated. The output in the CI4.4.6 and PHP 8.3.0 is as follows:

Screenshot 2024-03-14 165507

It is recommended to upgrade to the new version. Before anything else don't forget to make a backup then upgrade to the latest version step by step via:
https://codeigniter.com/user_guide/installation/upgrading.html

@kenjis I suggest moving the information shown in footer to the top, it can be helpful in reporting users.

Screenshot 2024-03-14 165748

**Comment by kenjis** [comment]
@datamweb I sent a PR #8626

**Comment by Nightprince** [comment]
> It seems you are using very old CI4. Could you update to v4.4.6?

problem has been fixed in latest version, but this pull request can be useful for those who use older versions.

**Comment by kenjis** [comment]
We recommend to use the latest version.

But if you really want to modify the deprecated method, modify it exactly the same way as the following change:
https://github.com/codeigniter4/CodeIgniter4/pull/8401/files
Or can I close this?

**Comment by Nightprince** [comment]
> We recommend to use the latest version.
> 
> But if you really want to fix the deprecated method, modify it exactly the same way as the following change:
> https://github.com/codeigniter4/CodeIgniter4/pull/8401/files
> Or can I close this?

Not important for me

**Pull request closed without merging**